### PR TITLE
Improve system-probe config loading

### PR DESCRIPTION
**Pull request by leeavital**
### What does this PR do?

* system_probe_config.enabled can be left unset, and will be implicitly
  enabled if any system_probe module is enabled
* network_config.enabled controls both the network module and connections check
* if network_config is not set, but system_probe.enabled, we will enable the system_probe implicitly to preserve back-compat
* EnabledSystemProbe if runtime_security_config.enabled is true

Also removes `system_probe_config.enabled` from template

@DataDog/networks 
@DataDog/agent-security 
@DataDog/container-integrations 

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

File: pkg/process/config/yaml_config.go [hunk 1]
```diff
@@ -176,24 +178,24 @@ func (a *AgentConfig) loadSysProbeYamlConfig(path string) error {
...
-		networkEnabled = config.Datadog.GetBool("network_config.enabled")
-		log.Info(fmt.Sprintf("network_config found, enabled = %v", networkEnabled))
-	} else {
-		log.Info("network_config not found, enabling network check by default")
-	}
-	if networkEnabled {
-		a.EnabledChecks = append(a.EnabledChecks, "Network")
+	// Enable network and connections check
+	if config.Datadog.GetBool("network_config.enabled") {
+		log.Info(fmt.Sprintf("network_config found.enabled found, enabling system-probe with network module enabled"))
```
Comment (stb2132) [inline pkg/process/config/yaml_config.go:183]: Grammar seems a little odd here, maybe:

> network_config found, enabled parameter set to true: enabling system-probe with network module running.

**Review by stb2132 — COMMENTED** [review]
LGTM, just one nit about logs

File: pkg/process/config/config_test.go [hunk 2]
```diff
@@ -232,7 +233,7 @@ func TestAgentConfigYamlAndSystemProbeConfig(t *testing.T) {
 	assert.Equal(10, agentConfig.QueueSize)
 	assert.Equal(true, agentConfig.AllowRealTime)
 	assert.Equal(true, agentConfig.Enabled)
-	assert.Equal(append(processChecks, "Network"), agentConfig.EnabledChecks)
+	assert.Equal(append(processChecks), agentConfig.EnabledChecks)
```
Comment (leeavital) [inline pkg/process/config/config_test.go:236]: The contents of `./testdata/TestDDAgentConfigYamlAndSystemProbeConfig.yaml` are

```
api_key: apikey_20
hostname: server-01

process_agent_enabled: true
process_config:
  process_dd_url: http://my-process-app.datadoghq.com
  enabled: 'true'
  queue_size: 10
  intervals:
    container: 8
    process: 30
  windows:
    args_refresh_interval: 100
    add_new_args: false
  scrub_args: false
  expvar_port: 5065

```

File: pkg/process/config/config_test.go [hunk 3]
```diff
@@ -290,7 +291,7 @@ func TestAgentConfigYamlAndSystemProbeConfig(t *testing.T) {
 	assert.Equal(1000, agentConfig.ClosedChannelSize)
 	assert.Equal(agentConfig.ExcludedBPFLinuxVersions, []string{"5.5.0", "4.2.1"})
 	assert.Equal("/var/my-location/system-probe.log", agentConfig.SystemProbeAddress)
-	assert.Equal(append(processChecks, "connections"), agentConfig.EnabledChecks)
+	assert.Equal(append(processChecks), agentConfig.EnabledChecks)
```
Comment (leeavital) [inline pkg/process/config/config_test.go:294]: The contents of `/testdata/TestDDAgentConfigYamlAndSystemProbeConfig-Net-2.yaml` are

```
system_probe_config:
    enabled: true
    sysprobe_socket: /var/my-location/system-probe.log
    disable_tcp: true
    disable_udp: true
    disable_ipv6: true
    disable_dns_inspection: false
    excluded_linux_versions:
      - 5.5.0
      - 4.2.1
    closed_channel_size: 1000
    source_excludes:
      127.0.0.1:
        - "5005"
      172.0.0.1/20:
        - "*"
      "*":
        - "443"
    dest_excludes:
      2001:db8::2:1:
        - "5005"
      172.0.0.1/20:
        - "*"
      "*":
        - "*"

network_config:
    enabled: false
```

File: pkg/process/config/config_test.go [hunk 4]
```diff
@@ -459,6 +460,47 @@ func TestEnvAdditionalEndpointsMalformed(t *testing.T) {
 	}
 }
 
+func TestNetworkConfig(t *testing.T) {
+	t.Run("yaml", func(t *testing.T) {
+		agentConfig, err := NewAgentConfig(
+			"test",
+			"./testdata/TestDDAgentConfigYamlOnly.yaml",
+			"./testdata/TestDDAgentConfig-NetConfig.yaml",
```
Comment (leeavital) [inline pkg/process/config/config_test.go:468]: the contents of `testdata/TestDDAgentConfig-NetConfig.yaml` are:

```
network_config:
  enabled: true
```

File: pkg/process/config/config_test.go [hunk 5]
```diff
@@ -459,6 +460,47 @@ func TestEnvAdditionalEndpointsMalformed(t *testing.T) {
...
+		assert.True(t, agentConfig.Enabled)
+		assert.ElementsMatch(t, []string{"connections", "Network", "process", "rtprocess"}, agentConfig.EnabledChecks)
+	})
+}
+
+func TestSystemProbeNoNetwork(t *testing.T) {
+	agentConfig, err := NewAgentConfig(
+		"test",
+		"./testdata/TestDDAgentConfigYamlOnly.yaml",
+		"./testdata/TestDDAgentConfig-OOMKillOnly.yaml",
```
Comment (leeavital) [inline pkg/process/config/config_test.go:494]: contents of `./testdata/TestDDAgentConfig-OOMKillOnly.yaml` are

```
system_probe_config:
  enable_oom_kill: true
```

**Review by heyronhay — COMMENTED** [review]
Do we want to switch the config template to default false for network config now with the new logic you added?
https://github.com/DataDog/datadog-agent/pull/6640/files#diff-a77843a3a2d99bdc21000210a1c6a0fc99871774b33d15b6779389e8299aa794R924-R932

File: pkg/config/config_template.yaml [hunk 6]
```diff
@@ -909,12 +909,6 @@ api_key:
 ## Uncomment this parameter and the one below to enable them.
 #
 # system_probe_config:
-
```
Comment (sunhay) [inline pkg/config/config_template.yaml:912]: Could you remove the `# Beta Agent` line above as well

File: pkg/process/config/yaml_config.go (hunk 1 above)
Comment (leeavital) [reply pkg/process/config/yaml_config.go:183]: I changed it to `network_config.enabled was true: enabling system-probe with network module running.`

**Comment by leeavital** [comment]
@heyronhay good point, done!

File: pkg/config/config_template.yaml [hunk 7]
```diff
@@ -935,7 +928,7 @@ api_key:
   ## @param enabled - boolean - optional - default: true
```
Comment (sunhay) [inline pkg/config/config_template.yaml:928]: `default: false` on this line

**Review by apigirl — APPROVED** [review]
👍 from docs

File: pkg/process/config/yaml_config.go [hunk 8]
```diff
@@ -159,13 +154,20 @@ func (a *AgentConfig) loadSysProbeYamlConfig(path string) error {
 	}
 
 	if config.Datadog.GetBool(key(spNS, "enable_tcp_queue_length")) {
+		a.EnableSystemProbe = true
```
Comment (heyronhay) [inline pkg/process/config/yaml_config.go:157]: Should we emit a log message as with security and network?

File: pkg/process/config/yaml_config.go [hunk 9]
```diff
@@ -159,13 +154,20 @@ func (a *AgentConfig) loadSysProbeYamlConfig(path string) error {
...
 		a.EnabledChecks = append(a.EnabledChecks, "TCP queue length")
 	}
 
 	if config.Datadog.GetBool(key(spNS, "enable_oom_kill")) {
+		a.EnableSystemProbe = true
```
Comment (heyronhay) [inline pkg/process/config/yaml_config.go:162]: Should we emit a log message as with security and network?

Comment (leeavital) [reply pkg/process/config/yaml_config.go:164]: I don't see why not! Done

File: pkg/process/config/yaml_config.go (hunk 8 above)
Comment (leeavital) [reply pkg/process/config/yaml_config.go:158]: Replied below, this is done!

File: pkg/config/config_template.yaml (hunk 7 above)
Comment (leeavital) [reply pkg/config/config_template.yaml]: Good catch, done

**Review by heyronhay — APPROVED** [review]
Thanks for the mods!

**Review by kevinconaway — APPROVED** [review]

**Pull request merged**